SimPEG/Regularization.py

```diff
@@ -416,655 +563,1030 @@ def eval2Deriv(self, m, v=None):
-    norms = [0., 2., 2., 2.] # Values for norm on (m, dmdx, dmdy, dmdz)
-    cell_weights = 1.        # Consider overwriting with sensitivity weights
-    nModels = 1 # Number of models
+    def __init__(self, mesh, **kwargs):
+        assert(mesh.dim > 2), (
+            "Mesh must have at least 3 dimensions to regularize along the "
+            "z-direction"
+        )
+        super(Smooth_zz, self).__init__(
+            mesh=mesh, orientation='z', **kwargs
+        )
 
-    def __init__(self, mesh, mapping=None, indActive=None, **kwargs):
-        Simple.__init__(self, mesh, mapping=mapping, indActive=indActive, **kwargs)
 
-        if isinstance(self.cell_weights,float):
-            self.cell_weights = np.ones(self.regmesh.nC) * self.cell_weights
+class Tikhonov(BaseComboRegularization):
+    """
+    L2 Tikhonov regularization with both smallness and smoothness (first order
+    derivative) contributions.
 
-        # if getattr(self, 'model', None) is None:
-        #             self.model = np.ones(self.regmesh.nC)
+    .. math::
+        \phi_m(\mathbf{m}) = \\alpha_s \| W_s (\mathbf{m} - \mathbf{m_{ref}} ) \|^2
+        + \\alpha_x \| W_x \\frac{\partial}{\partial x} (\mathbf{m} - \mathbf{m_{ref}} ) \|^2
+        + \\alpha_y \| W_y \\frac{\partial}{\partial y} (\mathbf{m} - \mathbf{m_{ref}} ) \|^2
+        + \\alpha_z \| W_z \\frac{\partial}{\partial z} (\mathbf{m} - \mathbf{m_{ref}} ) \|^2
 
-        # if self.regmesh.nC != len(self.model):
+    Note if the key word argument `mrefInSmooth` is False, then mref is not
+    included in the smoothness contribution.
 
-        #     nmod = len(self.model)/self.regmesh.nC
+    :param BaseMesh mesh: SimPEG mesh
+    :param IdentityMap mapping: regularization mapping, takes the model from model space to the thing you want to regularize
+    :param numpy.ndarray indActive: active cell indices for reducing the size of differential operators in the definition of a regularization mesh
+    :param bool mrefInSmooth: (default = False) put mref in the smoothness component?
+    :param float alpha_s: (default 1e-6) smallness weight
+    :param float alpha_x: (default 1) smoothness weight for first derivative in the x-direction
+    :param float alpha_y: (default 1) smoothness weight for first derivative in the y-direction
+    :param float alpha_z: (default 1) smoothness weight for first derivative in the z-direction
+    :param float alpha_xx: (default 1) smoothness weight for second derivative in the x-direction
+    :param float alpha_yy: (default 1) smoothness weight for second derivative in the y-direction
+    :param float alpha_zz: (default 1) smoothness weight for second derivative in the z-direction
+    """
 
-        #     assert np.mod(nmod, 1) > 1e-8, 'Mismatch between model and mesh'
+    def __init__(
+        self, mesh,
+        alpha_s=1e-6, alpha_x=1.0, alpha_y=1.0, alpha_z=1.0,
+        alpha_xx=Utils.Zero(), alpha_yy=Utils.Zero(), alpha_zz=Utils.Zero(),
```

Comment:
right now yes, but not in a safe way... 
- [ ] The combo regularization just doesn't include them in the list. it would be better to keep them in the `objfcts` list, but simply not evaluate. This can be fixed prior to merging